packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -222,13 +222,42 @@ abstract class SerializationManager {
     Object? object, {
     bool formatted = false,
   }) {
-    if (object is ProtocolSerialization) {
+    // detect List<? extends ProtocolSerialization>
+    if (object is List) {
+      object = _convertListForProtocol(object);
+    } else if (object is Map) {
+      // detect every ProtocolSerialization in map.values
+      object = _convertMapForProtocol(object);
+    } else if (object is ProtocolSerialization) {
       return encode(object.toJsonForProtocol(), formatted: formatted);
     }
-
     return encode(object, formatted: formatted);
   }
```

Comment:
Suggestion: Would the code be simpler to understand if we utilize a switch expression here? https://dart.dev/language/branches#switch-expressions